GTG/core/timer.py

```diff
@@ -82,3 +86,35 @@ def set_configuration(self, refresh_hour, refresh_min):
 
     def get_configuration(self):
         return self.config.get('hour'), self.config.get('min')
+
+    def time_parser(self, time):
+        hour_value = ''
+        minute_value = ''
+        # For clarity, I'm using two match criteria:
+        # Format for HH:MM and optional am/pm
+        match_1 = re.match('([0-9]{1,2})(:)([0-9]{2})( )?([ap]m)?', time)
+        # Format for HH and am/pm
+        match_2 = re.match('([0-9]{1,2})( )?([ap]m)', time)
```

Comment:
You can simplify this quite a bit. And there even 2 possible outcomes:

1) You create single regular expression to match all combinations, something like:

```
'([0-9]{1,2})(:)?([0-9]{2})?( )?([ap]m)?'
```

2) If you would like to preserve two separate regular expressions, you can parse them and pull out the important parts into variables, eg:

```python
if match_1:
  hour, minute, time = match_1.group(1), match_1.group(3), match_1, group(5)
```

And then do the validation only once.